src/poetry/console/commands/build.py

```diff
@@ -114,13 +114,22 @@ def _get_builder(self) -> Callable[..., None]:
 
         return self._build
 
+    def _has_build_backend_defined(self) -> bool:
+        return "build-backend" in self.poetry.file.read().get("build-system", {})
+
     def build(self, options: BuildOptions) -> int:
         if not self.poetry.is_package_mode:
             self.io.write_error_line(
                 "Building a package is not possible in non-package mode."
             )
             return 1
 
+        if not self._has_build_backend_defined():
```

Comment:
Won't this also be a surprise to users as this breaks previous behavior in the same way setuptools are being used? Not sure how many people leave out the section intentionally. 

That said, I'd also say that the current use of setuptools is in accordance with PEP518. 

> Tools should not require the existence of the [build-system] table. A pyproject.toml file may be used to store configuration details other than build-related data and thus lack a [build-system] table legitimately. If the file exists but is lacking the [build-system] table then the default values as specified above should be used. If the table is specified but is missing required fields then the tool should consider it an error.

Reply:
> Won't this also be a surprise to users as this breaks previous behavior in the same way setuptools are being used?

It will surprise people for sure. But now they get an error message. Without it `poetry build` would build a package, but it won't contain the data people expect. So "fail fast" is a better surprise ;)

Reply:
The problem is, how do you determine if they intended to use setuptools by relying on PEP518?

It is not unreasonable to imagine that Poetry is used for dev but the package needs setuptools for build.